EpsilonAuxBMS/Src/SetContactorsTask.c

```diff
@@ -0,0 +1,284 @@
+
+                if (!auxStatus.contactorError)
+                {
+                    // Turn on Common Contactor
+                    HAL_GPIO_WritePin(CONTACTOR_ENABLE1_GPIO_Port, CONTACTOR_ENABLE1_Pin, GPIO_PIN_SET);
+                    state = COMMON_CONTACTOR_CHECK;
+                }
+
+                break;
+
+            case COMMON_CONTACTOR_CHECK:
+                auxStatus.contactorError = checkIfContactorSet(SENSE1_Pin, SENSE1_GPIO_Port, 1);
+
+                if (!auxStatus.contactorError)
+                {
+                    contactorState = ON;
+                    // Turn on charge Contactor
+                    HAL_GPIO_WritePin(CONTACTOR_ENABLE2_GPIO_Port, CONTACTOR_ENABLE2_Pin, GPIO_PIN_SET);
+                }
+                else
+                {
+                    contactorState = OFF;
+                }
+
+                if (updateContactorState(contactorState, COMMON))
+                {
+                    state = CHARGE_CONTACTOR_CHECK;
+                }
+
+                break;
+
+            case CHARGE_CONTACTOR_CHECK:
+                auxStatus.contactorError = checkIfContactorSet(SENSE2_Pin, SENSE2_GPIO_Port, 2);
+
+                if (!auxStatus.contactorError)
+                {
+                    contactorState = ON;
+                    // Turn on discharge contactor
+                    HAL_GPIO_WritePin(CONTACTOR_ENABLE3_GPIO_Port, CONTACTOR_ENABLE3_Pin, GPIO_PIN_SET);
+                }
+                else
+                {
+                    contactorState = OFF;
+                }
+
+                if (updateContactorState(contactorState, CHARGE))
+                {
+                    state = DISCHARGE_CONTACTOR_CHECK;
+                }
+
+                break;
+
+            case DISCHARGE_CONTACTOR_CHECK:
+                auxStatus.contactorError = checkIfContactorSet(SENSE3_Pin, SENSE3_GPIO_Port, 3);
+
+                if (!auxStatus.contactorError)
+                {
+                    contactorState = ON;
+                    // Enable high voltage
+                    HAL_GPIO_WritePin(HV_ENABLE_GPIO_Port, HV_ENABLE_Pin, GPIO_PIN_SET);
+                }
+                else
+                {
+                    contactorState = OFF;
+                }
+
+                if (updateContactorState(contactorState, DISCHARGE))
+                {
+                    state = DONE;
+                }
+
+                break;
+
+            case DONE: ; // Put this here because compiler kept complaining about a declaration not being a statement
+                // could potentially cause problems.
+                uint8_t common = !HAL_GPIO_ReadPin(SENSE1_GPIO_Port, SENSE1_Pin);
+                uint8_t charge = !HAL_GPIO_ReadPin(SENSE2_GPIO_Port, SENSE2_Pin);
+                uint8_t discharge = !HAL_GPIO_ReadPin(SENSE3_GPIO_Port, SENSE3_Pin);
+
+                if (!common && !charge && !discharge) // None of the contactors are enabled
```

Comment:
For safety, we should just use `||` here instead of `&&`, reset everything is any of them are not set and start from the beginning. Then alot of the code below is not necessary.